File: Application/Students/GetStudentById/GetStudentByIdQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Biobrain.Application.Common.Core;
using Biobrain.Application.Common.Extensions;
using Biobrain.Application.Common.Specifications;
using Biobrain.Application.Interfaces.DataAccess;
using Biobrain.Application.Security;
using JetBrains.Annotations;
using Microsoft.EntityFrameworkCore;


namespace Biobrain.Application.Students.GetStudentById
{
    [PublicAPI]
    public sealed class GetStudentByIdQuery : IQuery<GetStudentByIdQuery.Result>
    {
        public Guid StudentId { get; init; }


        [PublicAPI]
        public record Result
        {
            public Guid StudentId { get; init; }
            public string Email { get; set; }
            public string FirstName { get; init; }
            public string LastName { get; init; }
            public string Country { get; init; }
            public string State { get; set; }
            public int? CurriculumCode { get; set; }
            public int? Year { get; set; }
            public Guid? SchoolId { get; init; }
            public List<Guid> SchoolClassIds { get; init; }
        }


        internal sealed class PermissionCheck : PermissionCheckBase<GetStudentByIdQuery>
        {
            private readonly IDb _db;

            public PermissionCheck(ISecurityService securityService, IDb db)
                : base(securityService)
                => _db = db;

            protected override bool CanExecute(GetStudentByIdQuery request, IUserSecurityInfo user)
            {
                if (user.IsApplicationAdmin())
                    return true;

				var student = _db.Students.Include(x => x.Schools).GetSingle(StudentSpec.ById(request.StudentId));
				if (student.Schools.Any(x => user.IsSchoolAdmin(x.SchoolId) || user.IsSchoolTeacher(x.SchoolId)))
					return true;

				return user.IsStudentAccountOwner(request.StudentId);
			}
        }


        internal sealed class Handler : QueryHandlerBase<GetStudentByIdQuery, Result>
        {
            public Handler(IDb db) : base(db) { }

            public override Task<Result> Handle(GetStudentByIdQuery request, CancellationToken cancellationToken)
            {
                return Db.Students
                         .Where(StudentSpec.ById(request.StudentId))
                         .Include(_ => _.User)
                         .Include(_ => _.SchoolClasses)
                         .Select(_ => new Result
                                      {
                                          StudentId = _.StudentId,
                                          Email = _.User.Email,
                                          FirstName = _.FirstName,
                                          LastName = _.LastName,
                                          Country = _.Country,
                                          State = _.State,
                                          Year = _.Year,
                                          CurriculumCode = _.CurriculumCode,
                                          //SchoolId = _.SchoolId,
                                          SchoolClassIds = _.SchoolClasses.Select(c => c.SchoolClassId).ToList(),
                                      })
                         .AsNoTracking()
                         .GetSingleAsync(cancellationToken);
            }
        }
    }
}

File: BiobrainWebAngular/src/app/api/students/get-student-by-id.query.ts
import { ApiPath } from '../api-path.service';
import { Query } from '../common/query';

export class GetStudentByIdQuery extends Query<GetStudentByIdQuery_Result> {
  constructor(public readonly studentId: string) {
    super();
  }

  getUrl(apiPath: ApiPath): string {
    return `${apiPath.students()}/GetById`;
  }

  deserializeResult(obj: GetStudentByIdQuery_Result_Object): GetStudentByIdQuery_Result {
    return new GetStudentByIdQuery_Result(
      obj.studentId,
      obj.email,
      obj.firstName,
      obj.lastName,
      obj.country,
      obj.state,
      obj.curriculumCode,
      obj.year,
      obj.schoolClassIds
    );
  }
}

export class GetStudentByIdQuery_Result {
  constructor(
    public readonly studentId: string,
    public readonly email: string,
    public readonly firstName: string,
    public readonly lastName: string,
    
    public readonly country: string,
    public readonly state: string,
    public readonly curriculumCode: number,
    public readonly year: number,

    public readonly schoolClassIds: string[]
  ) {
  }

  get fullName(): string {
    return `${this.firstName} ${this.lastName}`;
  }
}

export interface GetStudentByIdQuery_Result_Object {
  studentId: string;
  email: string;
  firstName: string;
  lastName: string;
  schoolClassIds: string[];
  country: string;
  state: string;
  curriculumCode: number;
  year: number;

}
